### PR TITLE
Add in-project resolution

### PR DESCRIPTION
**Pull request by hygt**

### CODE REVIEW

File: src/main/scala/ch/epfl/bluebrain/nexus/kg/resolve/InProjectResolution.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  * within a given project.
+  *
+  * @param project the resolution scope
+  * @param repo    the implicitly available resource repository
+  * @tparam F      the resolution effect type
+  */
+class InProjectResolution[F[_]: Functor](project: ProjectRef)(implicit repo: Repo[F]) extends Resolution[F] {
+
+  override def resolve(ref: Ref): F[Option[Resource]] = ref match {
+    case Latest(value)        => Resources.get(Id[ProjectRef](project, value)).value
```
Comment (bogdanromanx) [inline src/main/scala/ch/epfl/bluebrain/nexus/kg/resolve/InProjectResolution.scala]: the compiler should infer `Id[ProjectRef]` from `Id(project, value)`

File: src/main/scala/ch/epfl/bluebrain/nexus/kg/resolve/InProjectResolution.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+  * Simplest implementation that handles the resolution process of references to resources
+  * within a given project.
+  *
+  * @param project the resolution scope
+  * @param repo    the implicitly available resource repository
+  * @tparam F      the resolution effect type
+  */
+class InProjectResolution[F[_]: Functor](project: ProjectRef)(implicit repo: Repo[F]) extends Resolution[F] {
```
Comment (bogdanromanx) [inline src/main/scala/ch/epfl/bluebrain/nexus/kg/resolve/InProjectResolution.scala]: could you also add a companion object to construct the resolution instance without the new operator?
`InProjectResolution(project)`

File: src/main/scala/ch/epfl/bluebrain/nexus/kg/resolve/InProjectResolution.scala [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  override def resolveAll(ref: Ref): F[List[Resource]] =
+    resolve(ref).map(_.toList)
+}
+
+object InProjectResolution {
+
+  /**
+    * Constructs an [[InProjectResolution]] instance.
+    */
+  def apply[F[_]](project: ProjectRef)(implicit repo: Repo[F]): InProjectResolution[F] =
```
Comment (bogdanromanx) [inline src/main/scala/ch/epfl/bluebrain/nexus/kg/resolve/InProjectResolution.scala]: define it like so: `def apply[F[_] : Repo : Functor](project: ProjectRef): InProjectResolution[F] =`

**Comment by umbreak** [comment]
Test

**Pull request merged**